### DISCUSSION TITLE
fatal: Authentication failed for.git/'

### DISCUSSION THREAD

**Discussion by ulual** [community/community#51953] [New to GitHub] [answered]
Hello,

I need some help, I am setting up my GitHub account. I am sending the command "git push" and getting errors below. I tried resetting my password and did not help.

***
Victors-MacBook-Air:test ulua_libunao$ git push
Username for 'https://github.com': ulual
Password for 'https://ulual@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/ulual/test.git/'
Victors-MacBook-Air:test ulua_libunao$
***

**Comment by airtower-luna** [comment]
You cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by ulual** [reply]
generated access token and performed git push. Getting these errors below:

Victors-MacBook-Air:learner ulua_libunao$ git push
To https://github.com/ulual/learner.git
 ! [rejected]        main -> main (fetch first)
error: failed to push some refs to 'https://github.com/ulual/learner.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Reply by airtower-luna** [reply, marked as answer]
The message says exactly what's wrong: There are commits on the target branch that your push would overwrite. You have two options:

1. If you want to keep the commits from both sides, merge before pushing.
2. If you want to delete the remote-only commits, push with `--force`.

If you want to force-push double check first, or you might lose data. :warning:

**Reply by ulual** [reply]
Thank you all for all the great help. My issue was resolved by "git push --force".

**Comment by SMelidoni** [comment]
Hi there,

This error message indicates that the remote repository contains work that you do not have locally. This can happen if someone else has made changes to the remote repository since you last cloned it or pulled from it.

To resolve this issue, you will need to pull the changes from the remote repository before you can push your changes. You can do this by running the following command:

`git pull`

This will fetch the changes from the remote repository and merge them with your local repository. If there are any conflicts between your local changes and the changes from the remote repository, you will need to resolve them manually.

After resolving any conflicts, you can try pushing your changes again using the `git push` command.

If you want to overwrite the changes in the remote repository with your local changes, you can use the --force or -f flag with the git push command:

`git push --force`

However, be careful when using this flag, as it can potentially overwrite changes made by other contributors to the remote repository.

**Reply by ulual** [reply]
Thank you. It was fixed doing "git push --force"

**Reply by SMelidoni** [reply]
Great to hear! Would you mind marking it as the accepted solution so that others can easily find the solution to this problem? To do so, you can click on the 'Mark as solution' button underneath my answer. Thank you!
